Automattic/jetpack
# Fix: WooCommerce integration unit tests

#### Changes proposed in this Pull Request:
There are two primary goals for this PR:
1) Fix the ability to run WooCommerce integration unit tests. It seems that disuse of this feature resulted in the breakdown of the ability to run these tests over time.
2) Add a unit test that will catch future possible regressions such as the one added by [this change](https://github.com/Automattic/jetpack/pull/31423). Moving this action too early on the `init` hook resulted in Sync failing to sync WooCommerce content.

The changes will affect the following:
* Jetpack

#### Other information:


#### Jetpack product discussion
* p9dueE-7zI-p2

#### Does this pull request change what data or activity we track or use?
* No

#### Testing instructions:
1. Check out this PR's branch: `fix/jetpack-woocommerce-unit-tests`.
1. Remove the WooCommerce plugin from inside the Jetpack repo if you currently have a copy.
1. Download the 8.0.1 [release of WooCommerce](https://downloads.wordpress.org/plugin/woocommerce.8.0.1.zip) and move it to `projects/plugins/woocommerce` in the Jetpack repo.
1. Make a shallow clone of the WooCommerce repo: `git clone --depth 1 --branch 8.0.1 https://github.com/woocommerce/woocommerce`.
1. Move the `woocommerce/plugins/woocommerce/tests` directory from the cloned repo to `projects/plugins/woocommerce/` in the Jetpack repo.
1. Once you have the WooCommerce repo in place, run `jetpack docker phpunit-woocommerce -- --filter WP_Test_Jetpack_Sync_WooCommerce` to run just the Jetpack Sync + WooCommerce specific tests. You should see that 12 tests are executed with no failures and only 1 risky test.
